src/substrate/serviceRequestEvent.ts

```diff
@@ -52,4 +64,117 @@ export class ServiceRequestEventHandler {
       context
     )
   }
+
+  async insertStakingLoggingToDatabase(event) {
+    const serviceRequest = await event.data[1].toJSON();
+    const isServiceRequestHasBeenInsert = await this.loggingService.getLoggingByHashAndStatus(
+      serviceRequest.hash,
+      7
+    )
+    const stakingLogging : TransactionLoggingDto = {
+      address: serviceRequest.requesterAddress,
+      amount: Number(serviceRequest.stakingAmount)/ 10**18,
+      created_at: new Date(parseInt(serviceRequest.createdAt)),
+      currency: 'DBIO',
+      parent_id: BigInt(0),
+      ref_number: serviceRequest.hash,
+      transaction_status: 7,
+      transaction_type: 2
+    }
+    
+    if (!isServiceRequestHasBeenInsert) {
+      try {
+        await this.loggingService.create(stakingLogging)
+      } catch (error) {
+        await this.logger.log(error);
+      }
+    }
+  }
+
+  async insertWaitingForUnstakeLoggingToDatabase(event) {
+    const serviceRequest = await event.data[1].toJSON();
+    const serviceRequestParent = await this.loggingService.getLoggingByOrderId(
+      serviceRequest.hash
+    )
+    const isServiceRequestHasBeenInsert = await this.loggingService.getLoggingByHashAndStatus(
+      serviceRequest.hash,
+      11
+    )
+    const stakingLogging : TransactionLoggingDto = {
+      address: serviceRequest.requesterAddress,
+      amount: 0,
+      created_at: new Date(),
+      currency: 'DBIO',
+      parent_id: BigInt(serviceRequestParent.id),
+      ref_number: serviceRequest.hash,
+      transaction_status: 11,
+      transaction_type: 2
+    }
+
+    if(!isServiceRequestHasBeenInsert){
+      try {
+        await this.loggingService.create(stakingLogging)
+      } catch (error) {
+        await this.logger.log(error);
+      }
+    }
```

Comment:
What's `loggingService` for actually? You called `loggingService` before, the `try catch` should encompass the entirety of the method.

Also do this for:
- `insertUnstakeLoggingToDatabase`
- `insertWaitingForUnstakeLoggingToDatabase`
- `insertUnstakeLoggingToDatabase`
- `insertFinalizedLoggingToDatabase`

Reply:
okay, updated @agustinustheo